java/org/apache/catalina/session/StandardSession.java

```diff
@@ -107,6 +113,14 @@
         } else {
             LAST_ACCESS_AT_START = Boolean.parseBoolean(lastAccessAtStart);
         }
+
+        String principalSerializableCheck = System.getProperty(
+                "org.apache.catalina.session.StandardSession.PRINCIPAL_SERIALIZABLE_CHECK");
+        if (principalSerializableCheck == null) {
+            PRINCIPAL_SERIALIZABLE_CHECK = true;
+        } else {
+            PRINCIPAL_SERIALIZABLE_CHECK = Boolean.parseBoolean(principalSerializableCheck);
```

Comment:
System properties are nasty. Can we get rid of this, or implement it differently?